### DISCUSSION TITLE
Tracing cables which fork into multiple paths - what is workaround?

### DISCUSSION THREAD

**Discussion by dssdss** [netbox-community/netbox#7409] [Q&A]
Some of our fiber mediaconverters have one Gigabit ethernet port and two fiber ports - RX and TX. And I've emulated this part of my network in Netbox, and I've got Path split error, when I tried to trace it.

Jeremystretch said: "It's not a bug; what you've done with splitting out the Tx and Rx cables into separate paths isn't supported. You've told NetBox to trace from a single cable through a point where it's mapped to two separate cables, so the path split is expected. You can fix this by modeling each step along the path as a single cable."
I tried to trace single cable rx or tx, and I've got - No paths found. May be I'm doing something wrong? Can you help me, how can I trace singe cables along the path?
Details are below.

This is part of my network:
![image](https://user-images.githubusercontent.com/14805815/135459960-b3702fa9-0267-4883-b9ac-95f52a615711.png)

So, I've emulated this type of mediaconverter as a path panel (due to your advice in #3442 (comment) I've connected rear ge1 port with 2 fron ports - rx and tx):
![image](https://user-images.githubusercontent.com/14805815/135460015-cb47bee6-0c06-4dc2-beb4-3bc202789211.png)

But when I call Cable Tracing from my switch interface, I've got message: "Path split!"
![image](https://user-images.githubusercontent.com/14805815/135460271-3bc0ddd1-6c70-4280-8589-a91f88dda417.png)

Ok, I tried to trace single cable along the path. I opened my mediaconverter and tried to trace tx or rx cables:
![image](https://user-images.githubusercontent.com/14805815/135461890-89d90fd7-c64d-418b-b14e-89fdd0cf15c6.png)

And I've got - no paths found:
![image](https://user-images.githubusercontent.com/14805815/135462007-39b224b4-11ac-459c-a258-3f8a1dfac3d4.png)

Where is my mistake? How can I trace my links rx and tx?

**Comment by candlerb** [comment]
> Jeremystretch said:
> > "It's not a bug; what you've done with splitting out the Tx and Rx cables into separate paths isn't supported. You've told NetBox to trace from a single cable through a point where it's mapped to two separate cables, so the path split is expected. You can fix this by modeling each step along the path as a single cable."

He's right.

What he means is: replace the TX and RX ports on your media converters with a single "LC" port.  Drop the pairs of cables between the media converters, and replace them with a single cable.  In other words, treat the optical transmit/receive pair of fibres as a single cable, just as you treat the 8 copper wires in an RJ45 cable as a single cable.  Then path tracing will work fine.

The main problem with this approach is in modelling your patching infrastructure, if sometimes a path needs a single fibre (e.g. for GPON or Bidi optics), and sometimes it needs two fibres (for TX/RX).

The current workaround in Netbox is: connect the TX/RX fibre-pair cable to the left-hand patch panel port, and mark the adjacent patch panel port as "connected" (i.e. not available) - there's a flag for this on the port.

Alternatively, if you have *no* GPON or Bidi optics anywhere, and everything is TX/RX, then you can model your connections between patch panels such that one frontport represents a duplex pair of LC connectors.

**Reply by dssdss** [reply]
> What he means is: replace the TX and RX ports on your media converters with a single "LC" port. Drop the pairs of cables between the media converters, and replace them with a single cable. In other words, treat the optical transmit/receive pair of fibres as a single cable, just as you treat the 8 copper wires in an RJ45 cable as a single cable. Then path tracing will work fine.
Unfortunately, we can't replace RX and TX cables with one LC. All of our cables are singlemode. In my network picture you can see only 2 optic patch panels. In our real network there are may be 5-6 intermediate optic patch panels...

If I can't trace link between switches, is there possibility to separate trace RX and TX links at between all of my optic panesl (not only between neighboring patch panels, but to view all patch panels chain)? I still don't understand how to do this. 
All of my optic links on Netbox's patch panel have status linked on front and rear ports... If my explanations are not very clear, I can make more pictures...

**Reply by candlerb** [reply]
Try it.  Remodel the media converters with only *one* front port, linked to one rear port.  You'll be able to trace end-to-end between the switches; each media converter will appear just like another patch panel in the path.

But to take the second point: no, you won't be able to trace RX and TX paths separately.  The RX/TX path is traced as one "cable" even though it actually contains two fibres.  The comparison I made before is still valid: when you trace a CAT5e cable, you don't expect to trace all 8 copper wires separately.
